hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/util/Shell.java

```diff
@@ -86,7 +86,15 @@
    */
   @Deprecated
   public static boolean isJava7OrAbove() {
-    return true;
+    return isJavaSpecAtLeast(7);
+  }
+
+  // "1.8"->8, "9"->9, "10"->10
+  private static final int JAVA_SPEC_VER = Math.max(8, Integer.parseInt(
+      System.getProperty("java.specification.version").split("\\.")[0]));
+
+  public static boolean isJavaSpecAtLeast(int version) {
```

Comment:
1. Can we call this `isJavaVersionAtLeast` ? 
1. + javadoc explaining what it does
1. Needs a test in `TestShel` whichcalls the new API operation & sees that it is consistent with its expectations (i.e on trunk it must be 8+)

Reply:
Done all 3.